### PR TITLE
fix ios module with latest swift and xcode

### PR DESCRIPTION
**Pull request by akread**

### CODE REVIEW

File: ios/RNAdvertisingId.m [hunk 1]
```diff
@@ -8,6 +8,11 @@
 
 @interface RCT_EXTERN_MODULE(RNAdvertisingId, NSObject)
 
-RCT_EXTERN_METHOD(getAdvertisingId:(RCTResponseSenderBlock)callback)
+RCT_EXTERN_METHOD(getAdvertisingId:(RCTPromiseResolveBlock)resolve rejecter: (RCTPromiseRejectBlock)reject)
```
Comment (akread) [inline ios/RNAdvertisingId.m]: Here is the crux of the fix. I went to install this package on my react native application and I was getting `is not a recognized Objective-C method` errors on iOS. The swift and objective-c method signatures did not appear to align. Locally I fixed it by doing this and thought I would contribute it back.

After opening this repo locally I was unable to get the demo app running on Xcode 11 and iOS 13 which led to the other changes related to the swift version, bridging header files, package-lock.json file changes that came naturally when installing, and react native version update.

File: ios/RNAdvertisingId.m [hunk 2]
```diff
@@ -8,6 +8,11 @@
...
+
++ (BOOL)requiresMainQueueSetup
+{
+  return NO;
+}
```
Comment (akread) [inline ios/RNAdvertisingId.m]: If we don't explicitly disable main queue setup in this manner react native throws warnings about initializing on the main thread. This warning would appear for anyone using the package.

File: ios/RNAdvertisingId.swift [hunk 3]
```diff
@@ -14,7 +14,7 @@ class RNAdvertisingId: NSObject {
         let isAdvertisingTrackingEnabled : Bool = ASIdentifierManager.shared().isAdvertisingTrackingEnabled
         
         let response: NSMutableDictionary = [
-            "isLimitAdTrackingEnabled" : isAdvertisingTrackingEnabled,
+            "isLimitAdTrackingEnabled" : !isAdvertisingTrackingEnabled,
```
Comment (akread) [inline ios/RNAdvertisingId.swift]: Added this fix after further testing in my own application. This was reversed.

**Comment by FlorianBergmann** [comment]
Thanks for the merge request, we appreciate the support!

From my side it looks good but would like @15matias15 to check it too since my iOS knolwdege is rather limited.

**Comment by akread** [comment]
Sounds wonderful. Thanks!

**Comment by 15matias15** [comment]
@akread thanks for that PR, 
@FlorianBergmann  For me it's fine too.

**Comment by akread** [comment]
Thank you both! Is there anything else you would need from me to help test and release a new version?

**Comment by FlorianBergmann** [comment]
Unfortunately we are currently using this package in an Android only app so we are not able to test it much on our side. 

Do you mind me asking if you work as a react native developer and do you use the package in an production app? If that is the case and it works fine for you then I am okay with just merging it and releasing a new version.

Also we are always looking for new developers (we are based in Germany) and if you are interested, feel free to write me an email: florian@applike.info

**Comment by akread** [comment]
I do work on a react-native application that is publicly available on both iOS and Android platforms. We just added this library to our application which is where I ran into issues on iOS. Currently I have the dependency pinned to my commit and it has worked as expected in building and testing on our beta track. That said, the code which adds and uses the library has not yet been released in a production build.

If you prefer to wait until the build is in production I can let you know when the app version is released publicly.

**Comment by akread** [comment]
Any update on your thoughts here?

**Comment by FlorianBergmann** [comment]
Sorry for the late response. Since it seems to work fine for you and you also did your own testing I will just go ahead and merge it. I will then also publish a new version to npm.

**Pull request merged**